product_pack/views/product_product_views.xml

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 Tecnativa - Ernesto Tejeda
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+    <record model="ir.ui.view" id="nan_product_pack_form">
+        <field name="name">product.product.pack.form</field>
+        <field name="model">product.product</field>
+        <field name="inherit_id" ref="product.product_normal_form_view" />
+        <field name="arch" type="xml">
+            <field name="type" position="after">
+                <field name="pack"/>
+                <field name="pack_type" attrs="{'invisible':[('pack', '=', False)], 'required':[('pack','=',True)]}" context="{'pack_type': pack_type}"/>
+            </field>
+            <notebook position="inside">
+                <page string="Pack" attrs="{'invisible':[('pack', '=', False)]}">
```

Comment:
... and make the page always visible; just the fields in it become or not visible.